### ISSUE TITLE
Set::xor hangs

### ISSUE THREAD

**Issue by basz** [php-ds/extension#53]
Whilst trying to calculate a set of mutations the xor operation hangs

I expect $removalChangeSet to contain only ['member']

I'm using DS 1.0.4 extension on OSX
```
require 'vendor/autoload.php';

$roles             = new \Ds\Set(['guest', 'member']);
$requestedRemovals = new \Ds\Set(['member', 'nothing']);

$keepSet          = $roles->diff($requestedRemovals);
$removalChangeSet = $roles->xor($keepSet);

var_dump($removalChangeSet->toArray());
```

**Comment by basz** [comment]
update : using the `php-ds/php-ds` compatibility polypill no hang occurs

**Comment by rtheunissen** [comment]
Very interesting, the freezing bug was fixed at some point and doesn't exist in 1.1.4 (current version), but the `xor` doesn't produce the expected result. It's yielding `guest` as well.

**Comment by rtheunissen** [comment]
This has now been fixed in 3199f7893a911604399a0def58d3b86df23f1a40 and released as v1.1.5 on PECL.

**Comment by basz** [comment]
thanks!
